gtsam/nonlinear/Marginals.cpp

```diff
@@ -28,15 +28,35 @@ namespace gtsam {
 /* ************************************************************************* */
 Marginals::Marginals(const NonlinearFactorGraph& graph, const Values& solution, Factorization factorization,
                      EliminateableFactorGraph<GaussianFactorGraph>::OptionalOrdering ordering)
-{
+                     : values_(solution), factorization_(factorization) {
   gttic(MarginalsConstructor);
-
-  // Linearize graph
   graph_ = *graph.linearize(solution);
+  computeBayesTree(ordering);
+}
+
+/* ************************************************************************* */
+Marginals::Marginals(const GaussianFactorGraph& graph, const VectorValues& solution, Factorization factorization,
+                     EliminateableFactorGraph<GaussianFactorGraph>::OptionalOrdering ordering)
+                     : graph_(graph), factorization_(factorization) {
+  gttic(MarginalsConstructor);
+  Values vals;
+  for (const VectorValues::KeyValuePair& v: solution) {
```

Comment:
Let's modernize this to `for (const auto& keyValue: solution)` here and elsewhere in this file?

Reply:
done